docs/samples/Microsoft.ML.Samples/Dynamic/ValueMapping.cs

```diff
@@ -0,0 +1,281 @@
+        }
+
+
+        public static void ValueMappingTransform()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var ml = new MLContext();
+
+            // Get a small dataset as an IEnumerable.
+            IEnumerable<SamplesUtils.DatasetUtils.SampleInfertData> data = SamplesUtils.DatasetUtils.GetInfertData();
+            var trainData = ml.CreateStreamingDataView(data);
+
+            // Preview of the data.
+            //
+            // Age    Case  Education  induced     parity  pooled.stratum  row_num  ...
+            // 26.0   1.0   0-5yrs      1.0         6.0       3.0      1.0  ...
+            // 42.0   1.0   0-5yrs      1.0         1.0       1.0      2.0  ...
+            // 39.0   1.0   12+yrs      2.0         6.0       4.0      3.0  ...
+            // 34.0   1.0   0-5yrs      2.0         4.0       2.0      4.0  ...
+            // 35.0   1.0   6-11yrs     1.0         3.0       32.0     5.0  ...
+
+            StringToStringMappingExample(ml, trainData);
+            FloatToStringMappingExample(ml, trainData);
+            StringToKeyTypeMappingExample(ml, trainData);
+            StringToArrayMappingExample(ml, trainData);
+        }
+
+        ///<summary>
+        /// This example demonstrates the use of the ValueMappingEstimator by mapping string-to-string values. The ValueMappingEstimator uses
+        /// level of education as keys to a respective string label which is the value.
+        /// The mapping looks like the following:
+        /// <list>
+        ///     <item>0-5yrs -> Cat1</item>
+        ///     <item>6-11yrs -> Cat2</item>
+        ///     <item>12+yrs -> Cat3</item>
+        /// </list>
+        /// </summary>
+        public static void StringToStringMappingExample(MLContext ml, IDataView trainData)
+        {
+            // Creating a list of keys based on the Education values from the dataset
+            // These lists are created by hand for the demonstration, but the ValueMappingEstimator does take an IEnumerable.
+            var educationKeys = new List<string>()
+            {
+                "0-5yrs",
+                "6-11yrs",
+                "12+yrs"
+            };
+
+            var educationValues = new List<string>()
+            {
+                "Cat1",
+                "Cat2", 
+                "Cat3"
+            };
+
+            var pipeline = new ValueMappingEstimator<string, string>(ml, educationKeys, educationValues, ("Education", "EducationCategory"));
+
+            // The transformed data.
+            var transformedData = pipeline.Fit(trainData).Transform(trainData);
+
+            // Getting the data of the newly created column as an IEnumerable of SampleInfertDataWithFeatures.
+            var featuresColumn = ml.CreateEnumerable<SampleInfertDataWithFeatures>(transformedData, reuseRowObject: false);
```

Comment:
>featuresColumn [](start = 16, length = 14)

I find this name a bit confusing (since the data contains multiple columns). Perhaps featureRows? then it would be "foreach (featureRow in featureRows)".

Reply:
oh! good point.

---
In reply to: [250693972](https://github.com/dotnet/machinelearning/pull/2222#discussion_r250693972) [](ancestors = 250693972)